service-base/src/main/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecks.java

```diff
@@ -627,76 +647,146 @@ private Long extractLongValue(final JsonObject response, final Span span) {
+     * calculated as follows:
      * <pre>
      *             configured limit 
-     *   ---------------------------------- x No. of days from effectiveSince till lastDay of the targetDateMonth.
-     *    No. of days in the current month
+     *   ----------------------------------- x No. of minutes until start of next accounting period.
+     *   No. of minutes in the current month
      * </pre>
      * <p>
-     * For rest of the months and the <em>days</em> mode, the configured limit is used directly.
+     * In all other cases the configured limit is used directly.
      *
-     * @param effectiveSince The point of time on which the given resource limit came into effect.
-     * @param targetDateTime The target date and time.
-     * @param mode The mode of the period. 
-     * @param configuredLimit The configured limit. 
-     * @return The effective resource limit that has been calculated.
+     * @param effectiveSince The point of time (UTC) at which the resource limit became effective.
+     * @param targetDateTime The point in time (UTC) to calculate the limit for.
+     * @param periodType The type of accounting periods that the resource limit is based on.
+     * @param configuredLimit The maximum amount of resources to be used per accounting period.
+     * @return The resource limit for the current accounting period.
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     * @throws IllegalArgumentException if target date-time is before effective since date-time.
      */
     long calculateEffectiveLimit(
-            final OffsetDateTime effectiveSince,
-            final OffsetDateTime targetDateTime,
-            final PeriodMode mode,
+            final Instant effectiveSince,
+            final Instant targetDateTime,
+            final PeriodMode periodType,
             final long configuredLimit) {
-        if (PeriodMode.MONTHLY.equals(mode)
-                && configuredLimit > 0
-                && !targetDateTime.isBefore(effectiveSince)
-                && YearMonth.from(targetDateTime).equals(YearMonth.from(effectiveSince))
-                && effectiveSince.getDayOfMonth() != 1) {
-            final OffsetDateTime lastDayOfMonth = effectiveSince.with(TemporalAdjusters.lastDayOfMonth());
-            final long daysBetween = ChronoUnit.DAYS
-                    .between(effectiveSince, lastDayOfMonth) + 1;
-            return Double.valueOf(Math.ceil(daysBetween * configuredLimit / lastDayOfMonth.getDayOfMonth()))
-                    .longValue();
+
+        Objects.requireNonNull(effectiveSince, "effective since");
+        Objects.requireNonNull(targetDateTime, "target date-time");
+        Objects.requireNonNull(periodType, "period mode");
+
+        if (targetDateTime.isBefore(effectiveSince)) {
+            throw new IllegalArgumentException("target date-time must not be before effective since");
```

Comment:
FMPOV we should also allow to set a future date as an _effectiveSince_ date. For example, I would like to start a plan from the next month. In this case, I should be able to set it now and not wait until 12 AM UTC on the first day of the next month. IMHO if the effective date is in the future, we can simply ignore and no need to calculate any limits. 

If we don't want to allow any future date as an _effectiveSince_, then I think a corresponding validation should be  there while _adding/updating_ tenant information. WDYT?

Reply:
very good point :+1:  I will change it accordingly